lib/src/crawler.rb

```diff
@@ -0,0 +1,132 @@
+require 'open-uri'
+require 'src/interface_web'
+class Clawler
+  @page = Nokogiri::HTML(URI.open('https://careerbuilder.vn/viec-lam/tat-ca-viec-lam-vi.html'))
+
+  # PILL DATA CITIES
+  def self.make_cities
+    @data_list_cities = []        
+    data = @page.search("#location option")    
+    list_cities = data.to_s.split("</option>")
+    list_cities.each do |x|
+      @data_list_cities << x.gsub(/(^<[\w\D]*>)/, '').gsub(/\n/,'').rstrip
+    end
+
+    @data_list_cities.length.times do |i|
+      if i <= 69
+        name = (@data_list_cities[i].to_s)
+        City.create!(name: name, area: 1)  
+      elsif i > 69
+        name = (@data_list_cities[i].to_s)         
+        City.create!(name: name, area: 0)  
+      end
+    end
+  end
+    
+  #PIL DATA INDUSTRIES
+  def self.make_industries
+    @data_list_industries = []    
+    data = @page.search("#industry option")    
+    list_industries = data.to_s.split("</option>")
+    list_industries.each do |x|
+    @data_list_industries << x.gsub(/(^<[\w\D]*>)/, '').gsub(/\n/,'').strip
+    end
+    
+    @data_list_industries.length.times do |i|
+      name = (@data_list_industries[i].to_s)
+      if name.include?('&amp;')
+        name.gsub!('&amp;','&')
+      end
+      Industry.create!(name: name)  
+    end
+  end
+
+  # FILL DATA COMPANIES
+  def self.make_companies
+    Company.create!(name:     "Company name",
+                    address:    "Vui lòng xem trong mô tả công việc",
+                    short_description: "Vui lòng xem trong mô tả công việc")
+    @data = Interface_web.craw_data_companies()
+    puts 'Save info companies to database . . .'
+    i = @data[:name].length
+    i.times do |n|
```

Comment:
Có thể dùng các phương thức lặp `each, map` của Ruby để ngắn gon

```suggestion
    @data[:name].each do |n|
```